src/task.rs

```diff
@@ -1,17 +1,28 @@
 use ::std::collections::BTreeMap;
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde-support", derive(Serialize, Deserialize))]
 pub struct Task {
     pub subject: String,
     pub priority: u8,
+
+    #[cfg_attr(feature = "serde-support", serde(skip_serializing_if = "Option::is_none"))]
     pub create_date: Option<::Date>,
+
+    #[cfg_attr(feature = "serde-support", serde(skip_serializing_if = "Option::is_none"))]
     pub finish_date: Option<::Date>,
     pub finished: bool,
+
+    #[cfg_attr(feature = "serde-support", serde(skip_serializing_if = "Option::is_none"))]
     pub threshold_date: Option<::Date>,
+
+    #[cfg_attr(feature = "serde-support", serde(skip_serializing_if = "Option::is_none"))]
     pub due_date: Option<::Date>,
     pub contexts: Vec<String>,
     pub projects: Vec<String>,
     pub hashtags: Vec<String>,
+
+    #[cfg_attr(feature = "serde-support", serde(skip_serializing_if = "BTreeMap::is_empty"))]
```

Comment:
I don’t like the idea to remove empty fields.

On the client side (typically, a javascript front end) you need a special condition to handle the event of missing field. If the field is always available, you can do a for loop on an empty array.

It’s almost the same problem with removing options field. Using a ternary operator is more readable than an `if` condition with `undefined`.